Apache HttpClient requires new version in order to support SNI (Server Name Indication)

When downloading NodeJS from a HTTPS host I get the following error:
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:0.0.26:install-node-and-npm (install node and npm) on project uipatternlibrary-angularjs: Could not download Node.js: Could not download https://my.mirror.test.host/nodejs/v4.2.3/node-v4.2.3-linux-x64.tar.gz: hostname in certificate didn't match: <my.mirror.test.host> != <my.mirror.production.host> OR <my.mirror.production.host> OR <www.my.mirror.production.host> -> [Help 1]

As it turns out this is due to the fact that the used Apache HttpClient version (4.3.1) has no support for SNI (Server Name Indication).
The solution is simple... I just bumped "org.apache.httpcomponents:httpclient" to the latest version (4.5.1) and then everything works as expected.

--- a/frontend-plugin-core/pom.xml
+++ b/frontend-plugin-core/pom.xml
@@ -32,7 +32,7 @@
         <dependency>
             <groupId>org.apache.httpcomponents</groupId>
             <artifactId>httpclient</artifactId>
-            <version>4.3.1</version>
+            <version>4.5.1</version>
         </dependency>
 
         <dependency>

